nodejs/src/access-token.js

```diff
@@ -0,0 +1,137 @@
+const fetch = require('node-fetch');
+const { v4: uuidv4 } = require('uuid');
+
+class AccessToken {
+  constructor(envs) {
+    this.envs = envs;
+  }
+
+  async fetch() {
+    const req = new AccessTokenRequest(this.envs);
+    console.info(
+      'INFO: AccessToken.fetch - AccessTokenRequest',
+      req.url,
+      req.options,
+    );
+
+    return fetch(req.url, req.options)
+      .then((response) => {
+        console.info(
+          'INFO: AccessToken.fetch - response',
+          req.url,
+          response.status,
+          response.statusText,
+        );
+        if (response.status >= 400) {
+          const error = new AccessTokenResponseError(
+            req.url,
+            response.status,
+            response.statusText,
+          );
+          throw error;
+        }
+        return response.json();
+      })
+      .then((data) => {
```

Comment:
Do we need the extra then? Could just log and throw in the status check if block

Reply:
`response.json()` returns a `Promise`. You cannot directly access the returned JSON in the first `then`, so I think it's needed.

Reply:
gotcha.